web/engine/handlers.go

```diff
@@ -6,14 +6,40 @@ import (
 
 type HandlersS struct {
 	Method  string
-	Handler http.HandlerFunc
+	Handler http.Handler
 }
 
 type HandlersT map[string][]HandlersS
 
-func MakeHandler(m string, h http.HandlerFunc) HandlersS {
+func chain(h http.HandlerFunc, in ...func(http.HandlerFunc) http.HandlerFunc) http.HandlerFunc {
+	if len(in) == 0 {
+		return h
+	}
+
+	c := in[0]
+
+	if len(in) > 1 {
+		in = append(in[:0], in[1:]...)
+	} else {
+		in = nil
+	}
+
+	return chain(c(h), in...)
+
+}
+
+func reverse(in ...func(http.HandlerFunc) http.HandlerFunc) []func(http.HandlerFunc) http.HandlerFunc {
```

Comment:
And here it is not lack of golang background. The ability is not there. #go-nuts on freenode suggested doing a loop as the only other way to do it in the language is with the sort package and implementing  a sorter for the type which is more work then a single function.